carma_ros2_utils/test/test_carma_lifecycle_helpers.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <gtest/gtest.h>
+#include <memory>
+#include <chrono>
+#include <thread>
+#include <future>
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "carma_ros2_utils/carma_lifecycle_node.hpp"
+#include "std_srvs/srv/empty.hpp"
+#include "boost/core/ignore_unused.hpp"
+#include "boost/optional.hpp"
+using std_msec = std::chrono::milliseconds;
+
+namespace test_2
+{
+    class CarmaLifecycleNodeTest : public carma_ros2_utils::CarmaLifecycleNode
+    {            
+            
+        public:
+
+            int my_param_= 0;
+
+            int get_my_param()
+            {
+                return my_param_;
+            }
+
+            int set_my_param(int param)
+            {
+                int temp = my_param_;
+                my_param_ = param;
+                return temp;
```

Comment:
This looks like it will return the initialized value for my_param_ is that the intention?

Reply:
Yes the setter functions provided to the update_params method should return the old value.